packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -337,39 +344,44 @@ class EnrollVerbHandler extends AbstractVerbHandler {
     }
   }
 
-  Future<AtData> _fetchEnrollmentDataFromKeyStore(
-      String enrollmentKey, currentAtSign, String? enrollmentId) async {
-    AtData enrollData;
-    // KeyStore.get will not return null. If the value is null, keyStore.get
-    // throws KeyNotFoundException.
-    // So, enrollData will NOT be null.
-    try {
-      enrollData = await keyStore.get('$enrollmentKey$currentAtSign');
-    } on KeyNotFoundException {
-      throw AtEnrollmentException(
-          'enrollment id: $enrollmentId not found in keystore');
-    }
-    // If enrollment is not active, throw AtEnrollmentException
-    if (!SecondaryUtil.isActiveKey(enrollData)) {
-      throw AtEnrollmentException('The enrollment $enrollmentId is expired');
+  void _validateEnrollmentValidity(
+      EnrollStatus enrollStatus, String enrollmentId, Response response) {
+    if (EnrollStatus.expired == enrollStatus) {
+      response.isError = true;
+      response.errorCode = 'AT0028';
+      response.errorMessage =
+          'enrollment_id: $enrollmentId is expired or invalid';
     }
-    return enrollData;
   }
 
   /// Verifies whether the enrollment state matches the intended state.
   /// Throws AtEnrollmentException: If the enrollment state is different
   /// from the intended state.
   void _verifyEnrollmentStateBeforeAction(
```

Comment:
rename to _validateEnrollmentState

Reply:
changed as part of [4ef1efb](https://github.com/atsign-foundation/at_server/pull/1536/commits/4ef1efb0c490d5ec2459a5d7768fa917a0b28ab8)